src/index.ts

```diff
@@ -1,10 +1,16 @@
-// The gif
-export {Gif} from './Gif'
+export {Gif} from './Gif'; // (exported mostly for type checking)
+export {GifProvider} from './GifProvider';
+export {GifProviderBuilder} from './GifProviderBuilder';
 
-// The providers
-export {GifProvider} from './GifProvider'
-export {CompositeGifProvider, GiphyGifProvider, TenorGifProvider} from './providers';
-export {providersList} from './providers/providersList'
+// strategies
+export {
+    GifStrategy,
+    GiphyStrategy,
+    TenorStrategy
+} from './strategies';
 
-// The factory
-export {GifProviderFactory} from './GifProviderFactory'
+// clients
+export {
```

Comment:
You might want to export the `Client`s types so that is accessible to consumers.

```suggestion
export {
	TrendingClient,
	SearchClient,
```

Reply:
The included client's are for internal use only, and the strategy is the exposed interfacefor them.

TBH, the interfaces for the clients aren't really needed since they don't actually need to be standardized. They're more of a leftover from the earlier implemenation.

The flow is like this:
```
GifProvider -> (StrategyInterface) -> (Concrete)Strategy -> Client
```

So if the callee wants to create their own client, they can do with any interface, then wrap it in a Strategy.

Reply:
(I'm going to remove the `*Client` interfaces; see comment in thread below.)